import streamlit as st

st.set_page_config(page_title="Home")
st.title('About this website')

st.subheader("About epigenetics and association")
st.write("Epigenetics is the study of heritable changes in phenotypes not attributable to changes in DNA sequences, and epigenetic gene regulation plays a big role in regulating levels of gene expression, and is mediated by many processes, including DNA Methylation, Copy Number, and Mutation.")
st.write("DNA Methylation is the addition of a methyl CH3 group to a Cytosine in a CpG group; CpG is Cytosine-Guanine connected by a phosphate. Around 75% of CpGs are methylated in somatic cells, and unmethylated CpGs are typically clustered together in CpG islands, and 60-70% of human genes have a CpG island in their promoter region.")
st.write("When a promoter region gets methylated, transcription is inhibited, and aberrant methylation is a key sign of cancer. In typical tumor profiles, we note a global methylation trend, as most genes become differentially methylated from their matched normal counterparts. We measure methylation using probes of around 2 BP, where multiple probes may map to the same gene for greater coverage. Each probe has a Beta value indicating the intensity of methylation that ranges from 0 (not methylated) to 1 (methylated). CpG island methylation in promoter regions (200 BP upstream and 50 BP downstream of the start of the gene) is a well understood driver of altered gene expression, and increased methylation (hypermethylation) and decreased methylation (hypomethylation) of genes is a crucial measurement of what is impacting the gene.")
st.write("The data behind the website lies in association tables, which measure the association between gene expression and the epigenetic factors listed above to determine likely causes of gene expression. This is measured through a hypergeometric test to determine whether the representative samples for an epigenetic category are enriched in gene expression samples. The data all comes from TCGA, and p values have been corrected using the fdr correction method. Data entries on the tables reflect probes of genes, and any gene wide testing (gene expression vs CNV and Mutation) that doesn't rely on probe data is constant for all probes of the gene.")

st.subheader("Constructing the tables")
st.write("The initial probe data is obtained from the TCGA Methylation illumina 450k tables, where they exist. Outliers like Ovarian only have 27k probes. Next, these 450k probes have been shortlisted into ~35-40k by selecting the most highly variable probes of the dataset with regards to both tumor and normal samples, although work is being done to look into alternative methods to subset probes, such as considering highly variable probes amongst only tumor samples, and selecting only promoter region/CpG island based probes to make results more interpretable. Probe and gene location data is obtained using hg19, except for Ovarian data, whose probe locations are in hg18, and thus the locations are largely incorrect and ought to be ignored. There are 9 groups of note: High/Low gene expression (GT/GB), Hyper/Hypomethylation (MT/MB), Copy Number Gain/Deletion (CG/CD), and Mutation (Mut). For the pairs, representative samples have been chosen amongst the top/bottom 20% of samples depending on the category (high or low gene expression, etc), and Mutation is all samples with mutations in their gene. A hypergeometric test is then performed to determine association between these groups, and Bonferroni correction is then applied to adjust p values due to multiple hypothesis testing. Note that Gene expression, Methylation, CNV, and Mutation datasets all have a different set of samples, so we first standardize the samples such that all samples in our tables are in each dataset, so n in the hypergeometric test is standard. Furthermore, all aforementioned samples are tumor samples only, we filter out the normal samples before doing this testing. With the remaining normal samples isolated, we perform an identical analysis for gene expression vs methylation to obtain p values for the normal dataset. Then, when filtering for significantly associated probes, we filter out those probes that are significant in both tumor and normal data such that our results deal only with significant tumor probes.")

st.subheader("Data")
TCGA_data_URL = "https://xenabrowser.net/datapages/"
Association_Table_Data_URL = "https://github.com/jaypat3/MethylationPortal/tree/main/TCGADatasets"

st.subheader("All the files used on this website can be found here:")
st.write("TCGA data: [https://xenabrowser.net/datapages/](%s)" % TCGA_data_URL)
st.write("Association Table data: [https://github.com/jaypat3/MethylationPortal/tree/main/TCGADatasets](%s)" % Association_Table_Data_URL)